app/models/site.rb

```diff
@@ -7,10 +7,48 @@ class Site < ActiveRecord::Base
   accepts_nested_attributes_for :account, update_only: true
 
   class << self
-    delegate :account, :application_name, :institution_name, :institution_name_full, :reload, :update, to: :instance
+    delegate :account, :application_name, :institution_name,
+             :institution_name_full, :reload, :update, :announcement_text,
+             :marketing_text, :featured_researcher, to: :instance
 
     def instance
       first || create
     end
   end
+
+  def announcement_text
+    content_block_by_name(ContentBlock::ANNOUNCEMENT).value
+  end
+
+  def announcement_text=(value)
+    save_content_block_named(ContentBlock::ANNOUNCEMENT, value)
+  end
+
+  def marketing_text
+    content_block_by_name(ContentBlock::MARKETING).value
+  end
+
+  def marketing_text=(value)
+    save_content_block_named(ContentBlock::MARKETING, value)
+  end
+
+  def featured_researcher
+    content_block_by_name(ContentBlock::RESEARCHER).value
+  end
+
+  def featured_researcher=(value)
+    save_content_block_named(ContentBlock::RESEARCHER, value)
+  end
+
+  private
+
+    def save_content_block_named(name, value)
```

Comment:
How are you differentiating these on a per site basis?  It looks to me like there's only a single ContentBlock (per name) shared for all sites. Is that right?